src/notation/internal/notationmidiinput.cpp

```diff
@@ -140,8 +140,9 @@ void NotationMidiInput::doProcessEvents()
         for (const Note* note : notes) {
             notesItems.push_back(note);
         }
-
-        playbackController()->playElements(notesItems);
+        if (playbackConfiguration()->playNotesOnMidiInput()) {
+            playbackController()->playElements(notesItems);
```

Comment:
unfortunately that function (PlaybackController::playElements) then always looks at the other preference (play notes while editing) so if you have that off, MIDI input will never play back.